### ISSUE TITLE
Error: couldn't resolve module/action 'community.docker.docker_login'.

### ISSUE THREAD

**Issue by gowizzard** [ansible/ansible-lint#4046]
I have built a GitHub workflow and it should check the playbooks once, unfortunately I always get the following error:

```text
syntax-check[unknown-module]: couldn't resolve module/action 'community.docker.docker_login'. This often indicates a misspelling, missing collection, or incorrect module path.
Error: couldn't resolve module/action 'community.docker.docker_login'. This often indicates a misspelling, missing collection, or incorrect module path.
```

Here is the Ansible Lint GitHub workflow:

```yaml
name: Ansible Lint

on:
    pull_request:
        branches: [ "main" ]

jobs:
    lint:
        runs-on: ubuntu-latest

        steps:

            - name: Checkout repository
              uses: actions/checkout@v4

            - name: Run Ansible Lint
              uses: ansible/ansible-lint@main
```

And here is a test playbook:

```yaml
---
- name: Deploy Docker Compose
  hosts: nuc
  become: true
  vars_files:
    - secrets.yml

  tasks:
  - name: Copy Docker Compose file
    ansible.builtin.copy:
      src: ../compose.prod.yaml
      dest: /home/gowizzard/compose.yaml

  - name: Login to the Container registry
    community.docker.docker_login:
      registry: ghcr.io
      username: gowizzard
      password: "{{ github_token }}"
    no_log: true

  - name: Pull images
    ansible.builtin.command:
      chdir: /home/gowizzard
      cmd: docker compose pull

  - name: Run containers
    ansible.builtin.command:
      chdir: /home/gowizzard
      cmd: docker compose up -d

  - name: Remove old images
    ansible.builtin.command:
      chdir: /home/gowizzard
      cmd: docker system prune -f

```

Does anyone know the reason for this and how to fix it?
